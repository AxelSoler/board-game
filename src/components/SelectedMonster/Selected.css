.monsterAbout {
  width: 90%;
  margin: 24px auto;
  text-align: left;
}

p {
  font-family: 'Gill Sans', sans-serif;
}

.weaknesses {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  text-align: center;
}

.stars {
  padding: 0;
}

.weakness {
  margin: 4px auto;
  width: 33%;
}

h3 {
  text-align: center;
}
