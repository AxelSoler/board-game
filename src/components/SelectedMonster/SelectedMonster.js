import React from 'react';
import { useSelector } from 'react-redux';
import uniqid from 'uniqid';

const SelectedMonster = () => {
  const allMonsters = useSelector((state) => state);

  if (allMonsters.length !== 0) {
    const viewMonster = allMonsters[allMonsters.length - 1][0];
    return (
      <div>
        <h2>{viewMonster.monsterName}</h2>
        <p>{viewMonster.monsterDescription}</p>
        <p>{viewMonster.monsterSpecies}</p>
        <p>{viewMonster.monsterType}</p>
        {viewMonster.monsterWeaknesses.map((weakness) => (
          <div key={uniqid()}>
            <p>{weakness.element}</p>
            <p>{weakness.stars}</p>
          </div>
        ))}
      </div>
    );
  }

  return (
    <div>
      <p>return to main page</p>
    </div>
  );
};

export default SelectedMonster;
