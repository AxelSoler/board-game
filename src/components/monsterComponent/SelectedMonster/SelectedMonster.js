import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router';
import { NavLink } from 'react-router-dom';
import uniqid from 'uniqid';
import './Selected.css';
import { AiFillStar } from 'react-icons/ai';
import Header from '../../Header/Header';

const SelectedMonster = () => {
  const { monstername } = useParams();
  const allMonsters = useSelector((state) => state.allMonsters);
  const monsterArray = allMonsters.filter((element) => element.monsterName === monstername);
  const monster = monsterArray[0];

  if (monster !== undefined) {
    return (
      <div>
        <Header name={monster.monsterName} />
        <div className="monsterAbout">
          <img className="monsterImg" src={`/img/monsters/${monster.monsterName}.jpg`} alt={`${monster.monsterName}`} />
          <h2>{monster.monsterName}</h2>
          <div>
            <p>DESCRIPTION</p>
            <p>{monster.monsterDescription}</p>
          </div>
          <p>
            SPECIES:
            {' '}
            {monster.monsterSpecies}
          </p>
          <p>
            TYPE:
            {' '}
            {monster.monsterType}
          </p>
          <h3>WEAKNESS</h3>
          <div className="weaknesses">
            {monster.monsterWeaknesses.map((weakness) => (
              <div className="weakness" key={uniqid()}>
                <p>{weakness.element.toUpperCase()}</p>
                <ul className="stars">
                  {(() => {
                    const rows = [];
                    for (let i = 0; i < weakness.stars; i += 1) {
                      rows.push(<AiFillStar size="1.5em" color="yellow" key={i} />);
                    }
                    return rows;
                  })()}
                </ul>
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }

  return (
    <div>
      <button className="returnBtn" type="button">
        <NavLink to="/">RETURN TO MAIN PAGE</NavLink>
      </button>
    </div>
  );
};

export default SelectedMonster;
