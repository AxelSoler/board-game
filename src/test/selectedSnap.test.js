import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import SelectedMonster from '../components/monsterComponent/SelectedMonster/SelectedMonster';

test('selected snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <SelectedMonster />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
