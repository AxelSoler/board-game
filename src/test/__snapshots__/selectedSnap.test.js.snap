// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selected snapshot 1`] = `
<div>
  <p>
    return to main page
  </p>
</div>
`;
